qa/rpc-tests/cfund-paymentrequest-state-reorg.py

```diff
@@ -0,0 +1,111 @@
+        self.nodes = []
+        self.nodes.append(start_node(0, self.options.tmpdir, ["-debug","-headerspamfiltermaxsize=1000"]))
+        self.nodes.append(start_node(1, self.options.tmpdir, ["-debug","-headerspamfiltermaxsize=1000"]))
+        connect_nodes(self.nodes[0], 1)
+        self.is_network_split = False
+
+    def run_test(self):
+        self.nodes[0].staking(False)
+        self.nodes[1].staking(False)
+        activate_cfund(self.nodes[0])
+        self.sync_all()
+
+        self.nodes[0].donatefund(100)
+
+        # Generate our addresses
+        node_0_address = self.nodes[0].getnewaddress()
+        node_1_address = self.nodes[1].getnewaddress()
+
+        # Split funds
+        self.nodes[0].sendtoaddress(node_1_address, 5000000)
+        proposal=self.nodes[0].createproposal(node_0_address, 100, 36000, "test")
+        proposal_id=proposal["hash"]
+
+        slow_gen(self.nodes[0], 1)
+        end_cycle(self.nodes[0])
+
+        self.sync_all()
+
+        self.nodes[0].proposalvote(proposal_id, "yes")
+
+        slow_gen(self.nodes[0], 1)
+        end_cycle(self.nodes[0])
+
+        self.sync_all()
+
+        assert(self.nodes[0].getproposal(proposal_id)["state"] == 1)
+        assert(self.nodes[0].getproposal(proposal_id)["status"] == "accepted")
+
+        assert(self.nodes[1].getproposal(proposal_id)["state"] == 1)
+        assert(self.nodes[1].getproposal(proposal_id)["status"] == "accepted")
+
+        raw_preq = self.nodes[0].createpaymentrequest(proposal_id, 100, "preq", True)
+        self.sync_all()
+
+        # Disconnect Nodes 0 and 1
+        url = urllib.parse.urlparse(self.nodes[1].url)
+        self.nodes[0].disconnectnode(url.hostname+":"+str(p2p_port(1)))
+
+        self.nodes[0].forcetransactions([raw_preq])
+        self.nodes[1].forcetransactions([raw_preq])
+
+        blockcount_0 = self.nodes[0].getblockcount()
+        blockcount_1 = self.nodes[1].getblockcount()
+
+        self.nodes[0].staking(True)
+        self.nodes[1].staking(True)
+
+        # Let's wait for at least 3 blocks from Node 0
+        while self.nodes[0].getblockcount() - blockcount_0 < 3:
+            time.sleep(1)
+
+        # Node 1 only has 1 output so it will only stake 1 block
+        while self.nodes[1].getblockcount() == blockcount_1:
+            time.sleep(1)
+
+        print("nodes staked!")
+
+        self.nodes[0].staking(False)
+        self.nodes[1].staking(False)
+
+        node_0_best_hash = self.nodes[0].getblockhash(self.nodes[0].getblockcount())
+        node_1_best_hash = self.nodes[1].getblockhash(self.nodes[1].getblockcount())
+        node_0_nbits = int(self.nodes[0].getblock(node_0_best_hash)["bits"], 16)
+        node_1_nbits = int(self.nodes[1].getblock(node_1_best_hash)["bits"], 16)
+
+        # Node 0 and Node 1 have forked. Difficulty in Node 0 must be higher (hence lower nbits)
+        assert(node_0_best_hash != node_1_best_hash)
+        assert(node_0_nbits < node_1_nbits)
```

Comment:
This assertion is failing because the nbits are the same. If i comment out this assertion the rest of the tests pass. So the nodes start with different best block hashes and then they are the same. 

Seems to have started failing after merging #450 and #459 but i don't really know why that would effect this test.

Is the nbit check required?

Reply:
This means that some how the 2 nodes have the same best hash, even when they were staking while disconnected from each other.

How would that be possible?

Reply:
> This means that some how the 2 nodes have the same best hash, even when they were staking while disconnected from each other.
> 
> How would that be possible?

Scratch that, so they best hash are not the same, but the node_0_nbits assert is failing?

Reply:
Maybe this line should be :

```python
assert(node_0_nbits != node_1_nbits)
```

Reply:
> Maybe this line should be :
> 
> ```python
> assert(node_0_nbits != node_1_nbits)
> ```

The nbits might actually be the same, because of this change:

https://github.com/mxaddict/navcoin-core/blob/bc692d86e2360ed5ef10bc5d6cb1de3c3385bb88/src/chainparams.cpp#L469

Reply:
Yeah, the hashes are different but the nbits are the same. im reverting the target time to 30 seconds and seeing if it makes a difference.

Reply:
Good thinking @mxaddict the test passes with the nTargetSpacing=30.

So i would assume its okay to remove the assertion that the nbits are different after reducing the devnet blocktime to 5s since it seems its just a function of having such a short block time.

Will wait for @aguycalled to confirm its okay to remove the assertion then I'm happy with this PR.

Reply:
opted for remove that check